eng/common/scripts/Submit-PullRequest.ps1

```diff
@@ -48,87 +48,69 @@ param(
   [Parameter(Mandatory = $false)]
   [string]$PRBody = $PRTitle,
 
-  [Parameter(Mandatory = $false)]
-  [string]$PRLabels
-)
-
-$headers = @{
-  Authorization = "bearer $AuthToken"
-}
+  [string]$PRLabels,
 
-$query = "state=open&head=${PROwner}:${PRBranch}&base=${BaseBranch}"
+  [string]$UserReviewers,
 
-function AddLabels([int] $prNumber, [string] $prLabelString, [array] $existingLabels)
-{
-  # Adding labels to the pr.
-  if (-not $prLabelString) {
-    Write-Verbose "There are no labels added to the PR."
-    return
-  }
+  [string]$TeamReviewers,
 
-  # Parse the labels from string to array
-  $prLabelArray = @($prLabelString.Split(",") | % { $_.Trim() } | ? { return $_ })
-  foreach ($label in $existingLabels) {
-    if ($prLabelArray -contains $label.name) {
-      continue
-    }
-    $prLabelArray += $label.name
-  }
-  $prLabelUri = "https://api.github.com/repos/$RepoOwner/$RepoName/issues/$prNumber"
-  $labelRequestData = @{
-    labels = $prLabelArray
-  }
-  try {
-    $resp = Invoke-RestMethod -Method PATCH -Headers $headers $prLabelUri -Body ($labelRequestData | ConvertTo-Json)
-  }
-  catch {
-    Write-Error "Invoke-RestMethod $prLabelUri failed with exception:`n$_"
-  }
+  [string]$Assignees
+)
 
-  $resp | Write-Verbose
-  Write-Host -f green "Label(s) [$prLabelArray] added to pull request: https://github.com/$RepoOwner/$RepoName/pull/$prNumber"
-}
+. "${PSScriptRoot}\common.ps1"
 
 try {
-  $resp = Invoke-RestMethod -Headers $headers "https://api.github.com/repos/$RepoOwner/$RepoName/pulls?$query"
+  $resp = List-GithubPullRequests -RepoOwner $RepoOwner -RepoName $RepoName `
+  -Head "${PROwner}:${PRBranch}" -Base $BaseBranch
 }
 catch { 
-  Write-Error "Invoke-RestMethod [https://api.github.com/repos/$RepoOwner/$RepoName/pulls?$query] failed with exception:`n$_"
+  LogError "List-GithubPullRequests failed with exception:`n$_"
   exit 1
 }
+
 $resp | Write-Verbose
 
 if ($resp.Count -gt 0) {
-    Write-Host -f green "Pull request already exists $($resp[0].html_url)"
+  try {
+    LogDebug "Pull request already exists $($resp[0].html_url)"
 
     # setting variable to reference the pull request by number
     Write-Host "##vso[task.setvariable variable=Submitted.PullRequest.Number]$($resp[0].number)"
-    AddLabels $resp[0].number $PRLabels $resp[0].labels
-}
-else {
-  $data = @{
-    title                 = $PRTitle
-    head                  = "${PROwner}:${PRBranch}"
-    base                  = $BaseBranch
-    body                  = $PRBody
-    maintainer_can_modify = $true
-  }
 
-  try {
-    $resp = Invoke-RestMethod -Method POST -Headers $headers `
-                              "https://api.github.com/repos/$RepoOwner/$RepoName/pulls" `
-                              -Body ($data | ConvertTo-Json)
+    Add-GithubIssueLabels -RepoOwner $RepoOwner -RepoName $RepoName -IssueNumber $resp.number `
```

Comment:
I assume we can do one call to Update-Githubissue here?

Reply:
Yes, we can. But the more I worked on it I realize its better to just make separate calls. Otherwise we will have to get the current set of labels and assignees, then process the set to remove duplicates before calling `Update-GithubIssue`

Reply:
Also I wonder about the wisdom of even calling this here because if the PR was already created it probably was created in a previous run of the pipeline that currently running so the labels and assignees will have already been set. it seems this calls will be redundant anyways. Maybe we should only add reviewers, assignees and labels if the PR was created as part of the current run.

Reply:
I've had similar thoughts and mostly felt like it might be redundant but not cause any real issue. However I'd be Ok with us not adding this in the case where the PR already exists.